Runtime/Spectator/SpectatorController.cs

```diff
@@ -1,47 +1,33 @@
 ﻿using System;
-using UnityEngine;
+using Innoactive.Creator.Core.Input;
+using UnityEngine.InputSystem;
 
 namespace Innoactive.Creator.UX
 {
     /// <summary>
     /// Controller for a spectator to toggle UI visibility.
     /// </summary>
-    public class SpectatorController : MonoBehaviour
+    public class SpectatorController : InputActionListener
     {
         /// <summary>
         /// Event fired when UI visibility is toggled.
         /// </summary>
         public event EventHandler ToggleUIOverlayVisibility;
 
-        protected virtual void Update()
+        protected void OnEnable()
         {
-            HandleInput();
+            RegisterInputEvent(ToggleOverlay);
         }
-
-        /// <summary>
-        /// Handles user input.
-        /// </summary>
-        protected virtual void HandleInput()
+        
+        protected void OnDisable()
         {
-            if (IsKeyPressed(SpectatorSettings.Instance.ToggleOverlay))
-            {
-                ToggleUIOverlayVisibility?.Invoke(this, new EventArgs());
-            }
+            UnregisterInputEvent(ToggleOverlay);
         }
-        
-        /// <summary>
-        /// Is the given <paramref name="key"/> pressed.
-        /// </summary>
-        /// <param name="key">Key to check</param>
-        /// <returns>True, if key is pressed.</returns>
-        /// <remarks>Use this for the old input system.</remarks>
-        protected bool IsKeyPressed(KeyCode key)
+
+        public bool ToggleOverlay(InputAction.CallbackContext value)
         {
-#if ENABLE_LEGACY_INPUT_MANAGER
-            return Input.GetKeyDown(key);
-#else
+            ToggleUIOverlayVisibility?.Invoke(this, new EventArgs());
             return false;
```

Comment:
what those this even mean? Toggle should either return the state after toggling or nothing at all

Reply:
this is a method, connected to an event called by the new input control which allows you to swallow the event instead of continuing calling all lower prioritized listener. So this is implemented as expected.